troop/views.py

```diff
@@ -0,0 +1,92 @@
+from .forms import CreateParticipantForm
+
+
+class OnlyTroopManagerMixin(AccessMixin):
+    """Verify that the current user is allowed to manage the troop."""
+
+    def dispatch(self, request, *args, **kwargs):
+        if not request.user.is_authenticated:
+            return self.handle_no_permission()
+
+        troop_number = kwargs.get("troop")
+        if not troop_number:
+            return self.handle_no_permission()
+
+        troop = request.user.troops.filter(number=troop_number).first()
+        if not troop:
+            return self.handle_no_permission()
+
+        setattr(request, "troop", troop)
+        return super().dispatch(request, *args, **kwargs)
+
+
+class IndexView(OnlyTroopManagerMixin, generic.TemplateView):
+    template_name = "troop/index.html"
+
+
+class IndexParticipantView(OnlyTroopManagerMixin, generic.ListView):
+    template_name = "participant/index.html"
+    context_object_name = "participant_list"
+
+    def get_queryset(self):
+        return self.request.troop.participant_set.order_by(
+            Participant.age_section_order(), "-is_leader", "last_name", "first_name",
+        ).all().annotate(attendance_count=Count('attendance'))
+
+
+class UpdateParticipantView(OnlyTroopManagerMixin, generic.UpdateView):
+    model = Participant
+    form_class = CreateParticipantForm
+
+    def get_queryset(self):
+        return self.request.troop.participant_set
+
+    def get_form_kwargs(self, **kwargs):
+        form_kwargs = super().get_form_kwargs(**kwargs)
+        form_kwargs["troop"] = self.request.troop
+        return form_kwargs
+
+    def get_success_url(self):
+        messages.add_message(self.request, messages.SUCCESS, _("participant.saved"))
+        kwargs = {"troop": self.request.troop.number}
+        return reverse("troop:participant.index", kwargs=kwargs)
+
+    def form_invalid(self, form):
+        messages.add_message(self.request, messages.ERROR, _("form.error"))
+
+        response = super().form_invalid(form)
+        response.status_code = 422  # Unprocessable Entity
+        return response
+
+
+class CreateParticipantView(OnlyTroopManagerMixin, generic.CreateView):
+    model = Participant
+    form_class = CreateParticipantForm
+
+    def get_form_kwargs(self, **kwargs):
+        form_kwargs = super().get_form_kwargs(**kwargs)
+        form_kwargs["troop"] = self.request.troop
+        return form_kwargs
+
+    def get_success_url(self):
+        messages.add_message(self.request, messages.SUCCESS, _("participant.saved"))
+
+        kwargs = {"troop": self.request.troop.number}
+        if self.request.POST.get("_addanother"):
+            return reverse("troop:participant.create", kwargs=kwargs)
+        return reverse("troop:participant.index", kwargs=kwargs)
+
+    def form_invalid(self, form):
+        messages.add_message(self.request, messages.ERROR, _("form.error"))
+
+        response = super().form_invalid(form)
+        response.status_code = 422  # Unprocessable Entity
```

Comment:
Because the request was correctly formatted (django was able to receive and decode it), but the data it contained was not acceptable.

https://httpstatuses.com/422
> The server understands the content type of the request entity (hence a 415 Unsupported Media Type status code is inappropriate), and **the syntax of the request entity is correct** (thus a 400 Bad Request status code is inappropriate) but was unable to process the contained instructions.

This status is used by Laravel (PHP framework) for instance.